core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -105,19 +105,21 @@ public String generateLoadBalancerEndpoint(Stack stack) {
     }
 
     public String getLoadBalancerUserFacingFQDN(Long stackId) {
-        Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId);
-        if (!loadBalancers.isEmpty()) {
-            LoadBalancer preferredLB = loadBalancers.stream()
-                .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
-                .findAny()
-                .orElse(loadBalancers.iterator().next());
-            if (StringUtils.isNotEmpty(preferredLB.getFqdn())) {
-                return preferredLB.getFqdn();
-            } else {
-                return loadBalancers.stream()
-                    .map(LoadBalancer::getFqdn)
-                    .filter(StringUtils::isNotEmpty)
-                    .findFirst().orElse(null);
+        if (stackId != null) {
```

Comment:
when can this be null? at first glance this doesn't seem like a valid scenario

Reply:
I was hitting an NPE in a unit test and added this check while I was tracking down the source. It was something else, but I left it in because I figured it couldn't hurt. But if it's not a valid scenario I'll put it back.